src/Components/Components/src/Rendering/ComponentState.cs

```diff
@@ -222,5 +228,38 @@ private void DisposeBuffers()
             ((IDisposable)CurrentRenderTree).Dispose();
             _latestDirectParametersSnapshot?.Dispose();
         }
+
+        internal bool RequiresAsyncDisposal() => Component is IAsyncDisposable;
```

Comment:
Since `Component` is already public, the caller could check `componentState.Component is IAsyncDisposable` without needing this extra method. Not a big deal of course.

Reply:
I've removed this, I kind of hate the parenthesis soup that ! and is make together, hopefully in the future we will write `is not` instead